### DISCUSSION TITLE
[help] Targetizing rendered plots for use in Quarto

### DISCUSSION THREAD

**Discussion by Kodiologist** [ropensci/targets#1121] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

My project uses `tarchetypes::tar_quarto` to render a manuscript. The manuscript creates some `ggplot2` plots in blocks with headers like `{r agreement_plot, dev='png', dpi=300}`, which works fine, but leads to slow rendering for the manuscript if the plot itself is slow to render. So I'd like to save the rendered PNGs (as opposed to the `ggplot2` objects that produce them) as targets, and then include them in the Quarto. What's the cleanest way to do this? I thought of writing to a file in the target and then referring to the filename from the Quarto, but then `targets` wouldn't be able to tell that the manuscript depends on the plot target, right?

**Comment by wlandau** [comment]
You could create file targets for those PNG files using `format = "file"` in `tar_target()`. Or you could define targets for those plot files more compactly using `tar_file()` from `tarchetypes`.

**Reply by Kodiologist** [reply]
So if say `/mydir/foo.png` is a file target, and the manunscript contains `![A plot](/mydir/foo.png)`, that suffices for `targets` to infer the dependency of the manuscript on the file target? I presumed that `targets` only checks for `tar_load` and `tar_read` calls.

**Reply by wlandau** [reply]
Please have a look at https://github.com/ropensci/tarchetypes/discussions/134.

**Reply by Kodiologist** [reply, marked as answer]
Thanks, it looks like the key thing I was missing was `knitr::include_graphics`, as in

```
{r example-figure, fig.cap = "Example figure"}
knitr::include_graphics(targets::tar_read(TARGET))
```
